### PR TITLE
[sovereign] Cross header requester meta/extended

### PR DESCRIPTION
**Pull request by mariusmihaic**
## Reasoning behind the pull request
- Sync extended header by hash when bootstrapping from last finalized cross chain header
## Proposed changes
- Inject `CrossHeaderRequester` interface into `syncHeadersByHash.go` to request extended/meta headers
## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Review by sasurobert — DISMISSED** [review]

File: epochStart/bootstrap/bootStrapSovereignShardProcessor.go [hunk 1]
```diff
@@ -286,3 +288,12 @@ func (sbp *sovereignBootStrapShardProcessor) createEpochStartInterceptorsContain
 
 	return interceptorsContainerFactory.Create()
 }
+
+func (bp *sovereignBootStrapShardProcessor) createCrossHeaderRequester() (updateSync.CrossHeaderRequester, error) {
+	extendedHeaderRequester, castOk := bp.requestHandler.(updateSync.ExtendedShardHeaderRequestHandler)
+	if !castOk {
+		return nil, fmt.Errorf("%w in sovereignBootStrapShardProcessor.createHeadersSyncer for extendedHeaderRequester", process.ErrWrongTypeAssertion)
```
Comment (axenteoctavian) [inline epochStart/bootstrap/bootStrapSovereignShardProcessor.go:295]:
```suggestion
		return nil, fmt.Errorf("%w in sovereignBootStrapShardProcessor.createCrossHeaderRequester for extendedHeaderRequester", process.ErrWrongTypeAssertion)
```

File: epochStart/bootstrap/bootStrapSovereignShardProcessor_test.go [hunk 2]
```diff
@@ -293,3 +293,20 @@ func TestBootStrapSovereignShardProcessor_createEpochStartInterceptorsContainers
 	require.Empty(t, allKeys)
 	require.Zero(t, fullContainer.Len())
 }
+
+func TestBootStrapSovereignShardProcessor_createHeadersSyncer(t *testing.T) {
```
Comment (axenteoctavian) [inline epochStart/bootstrap/bootStrapSovereignShardProcessor_test.go:297]:
```suggestion
func TestBootStrapSovereignShardProcessor_createCrossHeaderRequester(t *testing.T) {
```

File: epochStart/bootstrap/bootStrapSovereignShardProcessor.go (hunk 1 above)
Comment (mariusmihaic) [reply epochStart/bootstrap/bootStrapSovereignShardProcessor.go:295]: Right, refactored

File: epochStart/bootstrap/bootStrapSovereignShardProcessor_test.go (hunk 2 above)
Comment (mariusmihaic) [reply epochStart/bootstrap/bootStrapSovereignShardProcessor_test.go]: Right, refactored

**Review by axenteoctavian — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Pull request merged**